### ISSUE TITLE
Implementing `in-daylight-saving-time` flag

### ISSUE THREAD

**Issue by guybedford** [WebAssembly/wasi-clocks#68]
Exploring what would be required for the JS implementation, I'm not sure if it is easily possibly to implement this flag on top of JavaScripts date implementation as this data isn't currently available, and some consistency questions may come up with regards to its implementation.

I am by no means an expert on this topic though, but this was originally mentioned by @ptomato.

**Comment by yoshuawuyts** [comment]
Yeah, also given the feedback provided in https://github.com/WebAssembly/wasi-clocks/pull/61#issuecomment-2174491391, we may want to consider dropping that flag entirely.

Talked with @sunfishcode a little about this yesterday: basic timezone support was intended to follow what POSIX provides. But that's clearly showing its age. We should want to update it to more recent standards where possible, especially because we care about targeting platforms like the web.

**Comment by yoshuawuyts** [comment]
I've filed https://github.com/WebAssembly/wasi-clocks/issues/69 to more broadly discuss changes to the timezones API. It seems daylight savings might be challenging for different reasons too. Let's merge this into the conversation there?
